api/v1/storagecluster_types.go

```diff
@@ -117,9 +117,6 @@ type StorageClusterSpec struct {
 	// provisioned by the storagecluster controller to be used in
 	// storageDeviceSets section of the CR.
 	BackingStorageClasses []BackingStorageClass `json:"backingStorageClasses,omitempty"`
-	// DefaultStorageProfile is the default storage profile to use for
-	// the storagerequest as StorageProfile is optional.
-	DefaultStorageProfile string `json:"defaultStorageProfile,omitempty"`
```

Comment:
This is dangerous as we don't know if there are any deployments where this was set in the past. 
I am not sure what will happen If we apply such a breaking change to the StorageCluster CR in deployments where the field was set

Reply:
I tried to search a bit on this, And I found the set field would be pruned once the field is removed from the CRD. Is there any other kind of risk you forsee with this?

Reply:
I did a small test to confirm this behaviour, and indeed the removed field from a CRD just gets pruned from the CR. Apart from that I don't see any errors or problems.
Steps taken-
1. Set the now unused sc.spec.version field to a value
2. Remove that field from the storageCluster CRD
3. Now check the StorageCluster spec.
4. The field is gone, no errors visible on the CR or the pods.

Reply:
https://ibm-systems-storage.slack.com/archives/C066ACYTR0Q/p1745929555313809?thread_ts=1745484815.003719&cid=C066ACYTR0Q